src/Mod/Arch/archobjects/base_rebar.py

```diff
@@ -0,0 +1,256 @@
+        the base rebar will be made for.
+    MarkNumber : App::PropertyInteger
+        Needed to identify reinforcements on the bar shape cut list.
+
+        Unique number on one bar shape cut list for manufactor
+        or for the building site. But for quantity report to know
+        how much material (kg steel) was used a bar shape cut list
+        could be made for a whole building. In this list duplicate
+        MarkNumber would be allowed. The user should be able
+        to explicit decide if in a rebar shape cut list
+        duplicate MarkNumbers are allowed.
+
+        May be a String should be used as MarkNumber. With this
+        MarkNumber could be integer, big or small ASCII character
+        or even Rome letter. But than another property is needed
+        to set the type mentioned above. But for what should it be set?
+        For each base rebar.
+    """
+
+    def __init__(
+        self,
+        obj
+    ):
+        super(BaseRebar, self).__init__(obj)
+        self.setProperties(obj)
+        obj.IfcType = "Undefined"  # only set in reinforcement
+        obj.setEditorMode("IfcType", 2)
+
+    def setProperties(
+        self,
+        obj
+    ):
+        self.Type = "RebarShape"
+        pl = obj.PropertiesList
+
+        # ArchComponent properties will be inherited
+        # Material, Base, MoveBase, ...
+        # base is a sketch or profil the ArchComponent is based on,
+        # for rebar the host is the father element,
+        # for rebar the structure the rebar is in
+
+        # do not show some not needed Component attributes in Editor
+        obj.setEditorMode("Additions", 2)
+        obj.setEditorMode("Axis", 2)
+        obj.setEditorMode("HorizontalArea", 2)
+        obj.setEditorMode("MoveWithHost", 2)
+        obj.setEditorMode("PerimeterLength", 2)
+        obj.setEditorMode("StandardCode", 2)  # useful,  not used ATM
+        obj.setEditorMode("Subtractions", 2)
+        obj.setEditorMode("VerticalArea", 2)
+        # CloneOf makes sense if two rebars are identical
+        # but should have different MarkNumber
+
+        # New properties
+        # Diameter
+        if "Diameter" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "Diameter",
+                "Rebar Shape",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "The diameter of this rebar shape"
+                )
+            )
+        # Rounding
+        if "Rounding" not in pl:
+            obj.addProperty(
+                "App::PropertyFloat",
+                "Rounding",
+                "Rebar",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The fillet to apply to the angle "
+                        "of the base profile. This value "
+                        "is multiplied by the bar diameter."
+                    )
```

Comment:
Additional parentheses are not needed, because the entire string is already inside the parentheses by `_NOOP`.
```py
QT_TRANSLATE_NOOP("App::Property",
                  "The fillet "
                  "something something "
                  "the end.")
```